### PR TITLE
[Task][74967] Create UI Movie Details & List Movie Screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 74967:  Create UI Movie Details & List Movie Screen](https://edu-redmine.sun-asterisk.vn/issues/74967)
## WHAT
- Create Ui Movie Detail Screen
- Create UI ListMovie by categories
- Logic click cell to Movie Detail screen in ListMovie
- Add logic save/remove movie in Favorite Screen
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: MovieHub/MovieHub/Scenes/Home/HomeViewController.swift [hunk 1]
```diff
@@ -170,6 +170,10 @@ extension HomeViewController: UITableViewDelegate, UITableViewDataSource {
                 movies = []
             }
             cell.configCell(movies: movies)
+            cell.tappedMovie = { [weak self] movie in
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/Home/HomeViewController.swift:173]: cái này sao ko dùng luôn delegate của tableview vậy em ?

File: MovieHub/MovieHub/Scenes/Home/HomeViewController.swift [hunk 2]
```diff
@@ -185,6 +189,13 @@ extension HomeViewController: UITableViewDelegate, UITableViewDataSource {
             let movieHeader = tableView.dequeueReusableHeaderFooterView(MovieHeader.self)
             guard let sectionType = HomeSectionType(rawValue: section) else { return nil }
             movieHeader?.configView(title: sectionType.title)
+            movieHeader?.showMoreTapped = { [weak self] in
+                guard let self = self else { return }
+                let vc = ListMovieViewController()
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/Home/HomeViewController.swift:194]: Khởi tạo từ Storyboard chứ nhỉ ?

File: MovieHub/MovieHub/Scenes/ListMovie/ListMovieViewController.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    
+    private func prepareDatasource() {
+        let categoryURL = Urls.shared.getListMovie(categories: category)
+
+        movieRepository.getMovies(urlString: categoryURL) { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success(let movieResponse):
+                DispatchQueue.main.async {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/ListMovie/ListMovieViewController.swift]: không cần dispatch main đoạn này

File: MovieHub/MovieHub/Scenes/MovieDetail/MovieDetailViewController.swift [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            let cell = tableView.dequeueReusableCell(for: indexPath, cellType: CastTableViewCell.self)
+            guard let movieDetail = movie else { return UITableViewCell() }
+            cell.prepareDatasource(data: movieDetail.credits?.cast ?? [])
+            cell.tappedCast = { [weak self] cast in
+                guard let self else { return }
+                self.toActorDetailScreen(cast: cast)
+            }
+            cell.selectionStyle = .none
+            return cell
+        case .similar:
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/MovieDetail/MovieDetailViewController.swift:93]: giữa các case thêm 1 break line nhìn cho clean

File: MovieHub/MovieHub/Scenes/Home/HomeViewController.swift (hunk 2 above)
Comment (ghost) [reply MovieHub/MovieHub/Scenes/Home/HomeViewController.swift:194]: Dạ không a  , em tạo xib ạ

**Review by dongtg-1485 — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**